src/function/statistics/corr.js

```diff
@@ -40,13 +44,32 @@ export const createCorr = /* #__PURE__ */ factory(name, dependencies, ({ typed,
    * @private
    */
   function _corr (A, B) {
-    if (Array.isArray(A[0]) && Array.isArray(B[0])) {
-      const correlations = []
+    const correlations = []
```

Comment:
The `_corr` function contains quite some duplicate code right now. I think changing it again to only handle `Array` and not `Matrix` will simplify it and remove this duplication. So, basically, take `_corr` as it was there and add the validations and handling of 2D there? Note that there is a util function `arraySize` that you can use if needed.

Reply:
Actually the code if for handling some cases like 1D array, 2D array and matrices. Changing it to previous code, will lose support for matrix.

Reply:
The behavior for Array and Matrix _should_ be exactly the same. And both Array and Matrix can hold 1D, 2D, or n-D content.

I think this already works in your implementation: I changed the implementation of the Matrix signature as follows, and then all unit tests pass, except for one (and in that case, the unit test wrongly expects an Array back but that should be a Matrix).

```js
    'Matrix, Matrix': function (A, B) {
      const res = _corr(A.toArray(), B.toArray())
      return Array.isArray(res) ? matrix(res) : res
    }
```

Does that make sense?

Reply:
yes absolutely.